File: mobile/constants/homeSpecialities.ts
import type { Ionicons } from "@expo/vector-icons";

export type HomeSpeciality = {
  name: string;
  filterKey: string;
  icon: keyof typeof Ionicons.glyphMap;
};

export const HOME_SPECIALITIES: HomeSpeciality[] = [
  { name: "Cardiology", filterKey: "cardiology", icon: "heart" },
  { name: "Orthopedics", filterKey: "orthopedics", icon: "body" },
  { name: "Psychiatry", filterKey: "psychiatry", icon: "happy" },
  { name: "Ophthalmology", filterKey: "ophthalmology", icon: "eye" },
  { name: "ENT", filterKey: "ent", icon: "ear" },
  { name: "Dentistry", filterKey: "dentistry", icon: "medical" },
  { name: "General Medicine", filterKey: "general medicine", icon: "medkit" },
  { name: "Gynecology", filterKey: "gynecology", icon: "female" },
  { name: "Dermatology", filterKey: "dermatology", icon: "color-palette" },
  { name: "Pediatrics", filterKey: "pediatrics", icon: "happy-outline" },
  { name: "Neurology", filterKey: "neurology", icon: "pulse" },
  { name: "Gastroenterology", filterKey: "gastroenterology", icon: "nutrition" },
];

File: flutter_mobile/lib/constants/home_specialities.dart
import 'speciality_assets.dart';

class HomeSpeciality {
  const HomeSpeciality({
    required this.name,
    required this.filterKey,
    required this.assetPath,
  });

  final String name;
  final String filterKey;
  final String assetPath;
}

/// Fixed 12 specialties — 4 columns × 3 rows on home dashboard.
List<HomeSpeciality> get homeSpecialities {
  const specs = [
    ('Cardiology', 'cardiology'),
    ('Orthopedics', 'orthopedics'),
    ('Psychiatry', 'psychiatry'),
    ('Ophthalmology', 'ophthalmology'),
    ('ENT', 'ent'),
    ('Dentistry', 'dentistry'),
    ('General Medicine', 'general medicine'),
    ('Gynecology', 'gynecology'),
    ('Dermatology', 'dermatology'),
    ('Pediatrics', 'pediatrics'),
    ('Neurology', 'neurology'),
    ('Gastroenterology', 'gastroenterology'),
  ];
  return specs
      .map(
        (s) => HomeSpeciality(
          name: s.$1,
          filterKey: s.$2,
          assetPath: SpecialityAssets.assetPathFor(s.$1) ?? '',
        ),
      )
      .where((s) => s.assetPath.isNotEmpty)
      .toList();
}
